internal/clisnapshot/snapshot.go

```diff
@@ -0,0 +1,104 @@
+package snapshot
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+)
+
+type Config struct {
+	// This Client points to a project's client, as set in the baseCommand
+	Client pb.WaypointClient
+}
+
+func (c *Config) WriteSnapshot(ctx context.Context, w io.Writer) error {
```

Comment:
I think I'd prefer just top-level functions here rather than hanging methods off a `Config`. I always find it a bit strange when a `Config` struct has methods unless its to manipulate/validate the config.

`func WriteSnapshot(ctx context.Context, client pb.WaypointClient, w io.Writer)` 

We can also use the option pattern for other args later if we wanted to. Not a huge deal to break compatibility here cause its an internal package.